external/fv3viz/fv3viz/_plot_cube.py

```diff
@@ -190,125 +209,45 @@ def plot_cube(
-            "coord_x_center": "x",
-            "coord_y_center": "y",
-            "coord_x_outer": "xb",
-            "coord_y_outer": "yb",
-            "coord_vars": {
-                "lonb": ["yb", "xb", "tile"],
-                "latb": ["yb", "xb", "tile"],
-                "lon": ["y", "x", "tile"],
-                "lat": ["y", "x", "tile"],
-            },
-        }
-
     Args:
         ds (xr.Dataset):
             Dataset containing the variable to be plotted, along with grid variables.
         var_name (str):
             Name of variable to be plotted.
-        coord_x_center (str):
-            name of the x-coordinate describing cell centers
-        coord_y_center (str):
-            name of the y-coordinate describing cell centers
-        coord_x_outer (str):
-            name of the x-coordinate describing cell interfaces
-        coord_y_outer (str):
-            name of the y-coordinate describing cell interfaces
-        coord_vars (Mapping[str, Sequence[str]]):
-            mapping of names of grid variables, which must include latitudes and
-            longitudes of both cell centers and bounds, to their sequence of
-            coordinate names
-
+        grid_metadata: vcm.cubedsphere.GridMetadata object describing dim
+            names and grid variable names
     Returns:
         ds (xr.Dataset):
             Dataset containing variable to be plotted as well as grid
-            coordinates variables, which are renamed and ordered for
-            plotting. Intended as first argument to :py:func:`plot_cube`.
-
-    Example:
-    ::
-        
-        # plot diag winds at two times
-        axes, hs, cbar = plot_cube(
-            mappable_var(diag_ds, 'VGRD850').isel(time = slice(2, 4)),
-            plotting_function = "contourf",
-            col = "time",
-            coastlines = True,
-            colorbar = True,
-            vmin = -20,
-            vmax = 20
-
-        )
+            variables, all of whose dimensions are ordered for plotting.
     """
-    for var, dims in coord_vars.items():
-        ds[var] = _remove_redundant_dims(ds[var], required_dims=dims)
-        ds[var] = ds[var].transpose(*dims)
-
-    first_dims = [coord_y_center, coord_x_center, "tile"]
-    rest = [dim for dim in ds[[var_name]].dims if dim not in first_dims]
-    xpose_dims = first_dims + rest
-    new_ds = ds[[var_name]].copy().transpose(*xpose_dims)
-
-    for grid_var in coord_vars:
-        new_ds = new_ds.assign_coords(coords={grid_var: ds[grid_var]})
-
-    for coord in [coord_y_center, coord_x_center, coord_y_outer, coord_x_outer]:
-        if coord in new_ds.coords:
-            new_ds = new_ds.drop(coord)
-
-    return new_ds
+    mappable_ds = xr.Dataset()
+    for var, dims in grid_metadata.coord_vars.items():
+        mappable_ds[var] = _align_grid_var_dims(ds[var], required_dims=dims)
+    var_da = _align_plot_var_dims(ds[var_name], grid_metadata.y, grid_metadata.x)
+    return mappable_ds.merge(var_da)
```

Comment:
Is this just a refactor or is there a more significant code change?

Reply:
Code change that accomplished 1) fixing the in-place modification of the coord vars bug and 2) no longer setting the grid vars as xarray coordinates since this was never actually needed by the `plot_cube` routine. Also allowed for adding more obvious error messages when the wrong grid metadata is passed and better tests.

Reply:
Okay thanks!